diff -rupN scipy-0.14.0/scipy/signal/tests/test_ltisys.py scipy-0.14.0_build/scipy/signal/tests/test_ltisys.py
--- scipy-0.14.0/scipy/signal/tests/test_ltisys.py	2014-02-04 05:56:48.000000000 +0100
+++ scipy-0.14.0_build/scipy/signal/tests/test_ltisys.py	2014-02-03 13:38:53.238617400 +0100
@@ -343,7 +343,7 @@ class Test_bode(object):
         # integrator, pole at zero: H(s) = 1 / s
         system = lti([1], [1, 0])
         w, mag, phase = bode(system, n=2)
-        assert_equal(w[0], 0.01)  # a fail would give not-a-number
+        assert_almost_equal(w[0], 0.01)  # a fail would give not-a-number
 
     def test_07(self):
         """bode() should not fail on a system with pure imaginary poles."""
@@ -434,7 +434,7 @@ class Test_freqresp(object):
         # integrator, pole at zero: H(s) = 1 / s
         system = lti([1], [1, 0])
         w, H = freqresp(system, n=2)
-        assert_equal(w[0], 0.01)  # a fail would give not-a-number
+        assert_almost_equal(w[0], 0.01)  # a fail would give not-a-number
 
     def test_from_state_space(self):
         # Ensure that freqresp works with a system that was created from the
 
