diff -rupN numpy-1.9.0rc1_org/numpy/core/src/multiarray/multiarraymodule.c numpy-1.9.0rc1/numpy/core/src/multiarray/multiarraymodule.c
--- numpy-1.9.0rc1_org/numpy/core/src/multiarray/multiarraymodule.c	2014-08-09 03:38:44.000000000 +0200
+++ numpy-1.9.0rc1/numpy/core/src/multiarray/multiarraymodule.c	2014-08-19 16:38:43.230012500 +0200
@@ -3994,10 +3994,7 @@ PyMODINIT_FUNC initmultiarray(void) {
 
 #if defined(MS_WIN64) && defined(__GNUC__)
   PyErr_WarnEx(PyExc_Warning,
-        "Numpy built with MINGW-W64 on Windows 64 bits is experimental, " \
-        "and only available for \n" \
-        "testing. You are advised not to use it for production. \n\n" \
-        "CRASHES ARE TO BE EXPECTED - PLEASE REPORT THEM TO NUMPY DEVELOPERS",
+        "Numpy 64bit experimental build with Mingw-w64 and OpenBlas.",
         1);
 #endif
 
diff -rupN numpy-1.9.0rc1_org/numpy/distutils/fcompiler/gnu.py numpy-1.9.0rc1/numpy/distutils/fcompiler/gnu.py
--- numpy-1.9.0rc1_org/numpy/distutils/fcompiler/gnu.py	2014-08-09 03:38:44.000000000 +0200
+++ numpy-1.9.0rc1/numpy/distutils/fcompiler/gnu.py	2014-08-19 16:53:44.266012500 +0200
@@ -21,6 +21,8 @@ TARGET_R = re.compile("Target: ([a-zA-Z0
 # XXX: handle cross compilation
 def is_win64():
     return sys.platform == "win32" and platform.architecture()[0] == "64bit"
+def is_win32():
+    return sys.platform == "win32" and platform.architecture()[0] == "32bit"
 
 if is_win64():
     #_EXTRAFLAGS = ["-fno-leading-underscore"]
@@ -35,6 +37,8 @@ class GnuFCompiler(FCompiler):
 
     def gnu_version_match(self, version_string):
         """Handle the different versions of GNU fortran compilers"""
+        if 'MinGW-W64' in version_string:
+            return ('gfortran', version_string.split('\n')[0].split(' ')[-1].strip())
         m = re.search(r'GNU Fortran', version_string)
         if not m:
             return None
@@ -129,7 +133,7 @@ class GnuFCompiler(FCompiler):
 
             opt.extend(['-undefined', 'dynamic_lookup', '-bundle'])
         else:
-            opt.append("-shared")
+            opt.append("-shared -Wl,-gc-sections -Wl,-s")
         if sys.platform.startswith('sunos'):
             # SunOS often has dynamically loaded symbols defined in the
             # static library libg2c.a  The linker doesn't like this.  To
@@ -199,9 +203,17 @@ class GnuFCompiler(FCompiler):
             # With this compiler version building Fortran BLAS/LAPACK
             # with -O3 caused failures in lib.lapack heevr,syevr tests.
             opt = ['-O2']
+        elif v and v>='4.6.0':
+            if is_win32():
+                # -mincoming-stack-boundary=2 needed
+                # due to the change to 16 byte stack alignment since GCC 4.6
+                # 32 bit Windows ABI defines 4 bytes stack alignment
+                opt = ['-O2 -march=i686 -mtune=generic -mfpmath=sse -msse2 -mincoming-stack-boundary=2'] 
+            else:
+                opt = ['-O2 -march=x86-64 -DMS_WIN64 -mtune=generic -msse2'] 
         else:
-            opt = ['-O3']
-        opt.append('-funroll-loops')
+            opt = ['-O2'] 
+        # opt.append()
         return opt
 
     def _c_arch_flags(self):
@@ -324,13 +336,11 @@ class Gnu95FCompiler(GnuFCompiler):
                     i = opt.index("gcc")
                     opt.insert(i+1, "mingwex")
                     opt.insert(i+1, "mingw32")
-            # XXX: fix this mess, does not work for mingw
+            # return opt in case of MinGW-W64
             if is_win64():
                 c_compiler = self.c_compiler
                 if c_compiler and c_compiler.compiler_type == "msvc":
                     return []
-                else:
-                    raise NotImplementedError("Only MS compiler supported with gfortran on win64")
         return opt
 
     def get_target(self):
@@ -344,10 +354,7 @@ class Gnu95FCompiler(GnuFCompiler):
         return ""
 
     def get_flags_opt(self):
-        if is_win64():
-            return ['-O0']
-        else:
-            return GnuFCompiler.get_flags_opt(self)
+        return GnuFCompiler.get_flags_opt(self)
 
 def _can_target(cmd, arch):
     """Return true is the command supports the -arch flag for the given
@@ -373,9 +380,13 @@ if __name__ == '__main__':
     from distutils import log
     log.set_verbosity(2)
 
-    compiler = GnuFCompiler()
-    compiler.customize()
-    print(compiler.get_version())
+    try:
+        compiler = GnuFCompiler()
+        compiler.customize()
+        print(compiler.get_version())
+    except Exception:
+        msg = get_exception()
+        print(msg)
 
     try:
         compiler = Gnu95FCompiler()
diff -rupN numpy-1.9.0rc1_org/numpy/distutils/mingw32ccompiler.py numpy-1.9.0rc1/numpy/distutils/mingw32ccompiler.py
--- numpy-1.9.0rc1_org/numpy/distutils/mingw32ccompiler.py	2014-08-09 03:38:44.000000000 +0200
+++ numpy-1.9.0rc1/numpy/distutils/mingw32ccompiler.py	2014-08-19 16:52:30.756012500 +0200
@@ -85,17 +85,28 @@ class Mingw32CCompiler(distutils.cygwinc
         elif self.linker_dll == 'gcc':
             self.linker = 'g++'
 
-        # **changes: eric jones 4/11/01
-        # 1. Check for import library on Windows.  Build if it doesn't exist.
-
-        build_import_library()
-
-        # Check for custom msvc runtime library on Windows. Build if it doesn't exist.
-        msvcr_success = build_msvcr_library()
-        msvcr_dbg_success = build_msvcr_library(debug=True)
-        if msvcr_success or msvcr_dbg_success:
-            # add preprocessor statement for using customized msvcr lib
-            self.define_macro('NPY_MINGW_USE_CUSTOM_MSVCR')
+        p = subprocess.Popen(['gcc', '--version'], shell=True,
+                             stdout=subprocess.PIPE)
+        out_string = p.stdout.read()
+        p.stdout.close()
+        
+        # Before build with MinGW-W64 generate the python import library
+        # with gendef and dlltool according to the MingW-W64 FAQ. 
+        # Use the MinGW-W64 provided msvc runtime import libraries.
+        # Don't call build_import_library() and build_msvcr_library.
+        
+        if 'MinGW-W64' not in str(out_string):
+            
+            # **changes: eric jones 4/11/01
+            # 1. Check for import library on Windows.  Build if it doesn't exist.            
+            build_import_library()
+            
+            # Check for custom msvc runtime library on Windows. Build if it doesn't exist.
+            msvcr_success = build_msvcr_library()
+            msvcr_dbg_success = build_msvcr_library(debug=True)
+            if msvcr_success or msvcr_dbg_success:
+                # add preprocessor statement for using customized msvcr lib
+                self.define_macro('NPY_MINGW_USE_CUSTOM_MSVCR')
 
         # Define the MSVC version as hint for MinGW
         msvcr_version = '0x%03i0' % int(msvc_runtime_library().lstrip('msvcr'))
@@ -124,10 +135,10 @@ class Mingw32CCompiler(distutils.cygwinc
             else:
                 # gcc-4 series releases do not support -mno-cygwin option
                 self.set_executables(
-                    compiler='gcc -g -DDEBUG -DMS_WIN64 -O0 -Wall',
-                    compiler_so='gcc -g -DDEBUG -DMS_WIN64 -O0 -Wall -Wstrict-prototypes',
-                    linker_exe='gcc -g',
-                    linker_so='gcc -g -shared')
+                    compiler='gcc -march=x86-64 -mtune=generic -DMS_WIN64 -O2 -msse2 -Wall',
+                    compiler_so='gcc -march=x86-64 -mtune=generic -DMS_WIN64 -O2 -msse2 -Wall -Wstrict-prototypes',
+                    linker_exe='gcc',
+                    linker_so='gcc -shared -Wl,-gc-sections -Wl,-s')
         else:
             if self.gcc_version <= "3.0.0":
                 self.set_executables(compiler='gcc -mno-cygwin -O2 -w',
@@ -142,10 +153,11 @@ class Mingw32CCompiler(distutils.cygwinc
                                      linker_so='g++ -mno-cygwin -shared')
             else:
                 # gcc-4 series releases do not support -mno-cygwin option
-                self.set_executables(compiler='gcc -O2 -Wall',
-                                     compiler_so='gcc -O2 -Wall -Wstrict-prototypes',
+                # i686 build needs '-mincoming-stack-boundary=2' due to ABI incompatibility to Win32 ABI
+                self.set_executables(compiler='gcc -O2 -march=i686 -mtune=generic -mfpmath=sse -msse2 -mincoming-stack-boundary=2 -Wall',
+                                     compiler_so='gcc -O2 -march=i686 -mtune=generic  -mfpmath=sse -msse2 -mincoming-stack-boundary=2 -Wall -Wstrict-prototypes',
                                      linker_exe='g++ ',
-                                     linker_so='g++ -shared')
+                                     linker_so='g++ -shared -Wl,-gc-sections -Wl,-s')
         # added for python2.3 support
         # we can't pass it through set_executables because pre 2.2 would fail
         self.compiler_cxx = ['g++']
diff -rupN numpy-1.9.0rc1_org/numpy/setup.py numpy-1.9.0rc1/numpy/setup.py
--- numpy-1.9.0rc1_org/numpy/setup.py	2014-08-09 03:38:44.000000000 +0200
+++ numpy-1.9.0rc1/numpy/setup.py	2014-08-19 16:38:43.400012500 +0200
@@ -22,6 +22,7 @@ def configuration(parent_package='',top_
     config.add_subpackage('doc')
     config.add_data_dir('doc')
     config.add_data_dir('tests')
+    config.add_data_files('core/libopenblas.dll') # copy libopenblas.dll into numpy/core
     config.make_config_py() # installs __config__.py
     return config
 
